frontend/apps/reader/modules/readermenusearch.lua

```diff
@@ -0,0 +1,118 @@
+    end
+    self.search_for = G_reader_settings:readSetting("menu_search_string", _("Help"))
+    self.animation_time_s = G_reader_settings:readSetting("menu_search_animation_time_s", 1.0)
+end
+
+function ReaderMenuSearch:addToMainMenu(menu_items)
+    menu_items.search_menu = {
+        text = _("Menu Search"),
+        callback = function()
+            local search_dialog
+            search_dialog = InputDialog:new{
+                title = _("Search menu entry"),
+                description = _([[Search for a menu entry containing the following text (case insensitive).
+
+Attention: Lua patterns are used. If you want to search for '%' or '.' you have to enter '%%' or '%.']]),
+                input = self.search_for,
+                buttons = {
+                    {
+                        {
+                            text = _("Cancel"),
+                            id = "close",
+                            callback = function()
+                                UIManager:close(search_dialog)
+                            end,
+                        },
+                        {
+                            text = _("OK"),
+                            callback = function()
+                                self.search_for = search_dialog:getInputText()
+                                G_reader_settings:saveSetting("menu_search_string", self.search_for)
+                                UIManager:close(search_dialog)
+                                self:hereWeGo(Utf8Proc.lowercase(self.search_for))
+                            end,
+                        },
+                    }
+                },
+            }
+
+            local check_button_animation = CheckButton:new{
+                text = _("Animation"),
+                checked = self.animation_time_s ~= 0.0,
+                parent = search_dialog,
+                callback = function()
+                    if self.animation_time_s == 0 then
+                        self.animation_time_s = 1.0
+                    else
+                        self.animation_time_s = 0.0
+                    end
+                    G_reader_settings:saveSetting("menu_search_animation_time_s", self.animation_time_s)
+                end,
+            }
+            search_dialog:addWidget(check_button_animation)
+
+            UIManager:show(search_dialog)
+            search_dialog:onShowKeyboard()
+        end,
+        keep_menu_open = true,
+    }
+end
+
+function ReaderMenuSearch:getCurrentSearchResults()
+    local function item_callback(i)
+        self.kv:onClose()
+        UIManager:setDirty(nil, "ui")
+
+        local path = TouchMenu.foundMenuItem[i][2]
+        UIManager:sendEvent(Event:new("OpenMenu", path, self.animation_time_s))
+    end
+    local function item_hold_callback(i)
+        UIManager:show(InfoMessage:new{
+            text = T(_("%1\n\n%2"), TouchMenu.foundMenuItem[i][1], TouchMenu.foundMenuItem[i][3]),
+        })
```

Comment:
You might want to add a `\n` before each `->` for better legibility (rather than have it wrapped inside a path at any random point).